src/components/LoginPage/LoginPageComponent.js

```diff
@@ -115,9 +117,7 @@ const Footer = styled.footer`
 
 export class LoginPageComponent extends Component {
   render() {
-    const authHandler = this.props.isGoogleApiInitialized
-      ? this.props.requestAuthorization
-      : null;
+    const authHandler = this.props.requestAuthorization;
```

Comment:
Why this change?

Reply:
As you can see in the code **authHandler** is on click handler. Generally you never assign handle function conditionally or at least I can't came up with use case where you will need to do so.

Also **isGoogleApiInitialized** is true when corresponding action is triggered, but the action itself changes no state. Basically it's more complicated way to **console.log("GAPI Initialized"), but the main reason is the reason above, that it serves no function except being a condition where it need not to be.

You can think that we need conditional assignment because we don't want button to use uninitialized code, but the **isGoogleApiInitialized** itself not connected to any code that should be initialized before using, it just exists independently

**requestAuthorization** function itself just makes the same, emitting another action, so it's safe to assign handler to button

Reply:
**requestAuthorization** is dispatching action, so it's safe to assign it without checking gapi initialization.

Reply:
Saga listens to this action and calls `gapiAuthorize()` which calls `gapi.auth.authorize()`. So if `gapi` is not initialized, it will fail with `can not call .auth() of undefined`.
How does dispatching an action help us to make it safe?

Reply:
If what you saying is true, the app would not work.
![Screenshot 2020-10-25 07:32:01](https://user-images.githubusercontent.com/45026388/97094901-4eec1e00-1694-11eb-8c6d-ab32c2a68cfe.png)

**gapiAuthorize** are called independently of **isGoogleApiInitialized**

Reply:
fixed by https://github.com/DarthVanger/NeuralNotes/pull/167/commits/2cf00e0e0593bd1f37ac43153229f3446b782603